scripts/tslint/enforcePrimitiveNamesRule.ts

```diff
@@ -0,0 +1,39 @@
+import * as Lint from "tslint/lib/lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static getReplacement(name: string): string {
+        let replacement: string;
+        switch (name) {
+            case "Number":
+                replacement = "number";
+                break;
+            case "Boolean":
+                replacement = "boolean";
+                break;
+            case "Object":
+                replacement = "{}";
+                break;
+            case "String":
+                replacement = "string";
+                break;
+        }
+        return replacement;
+    }
+    public static FAILURE_STRING_FACTORY = (name: string, replacement: string) => `Use of non primitive name '${name}' is not allowed - use '${replacement}' instead`;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new EnforcePrimitiveNames(sourceFile, this.getOptions()));
+    }
+}
+
+class EnforcePrimitiveNames extends Lint.RuleWalker {
+    visitIdentifier(node: ts.Identifier): void {
+        super.visitIdentifier(node);
+        if (node.text === "Number" || node.text === "Boolean" || node.text === "Object" || node.text === "String") {

```

Comment:
"Object" is too strong here if we do not check that it was not already defined, and or qualified. There are definitions with `Object` e.g.:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/should/should.d.ts#L6
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/chai/chai.d.ts#L412
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/sharepoint/SharePoint.d.ts#L1642